services/bonita-theme/bonita-theme-impl/src/main/java/org/bonitasoft/engine/theme/impl/ThemeActionCalculator.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (C) 2015 Bonitasoft S.A.
+ * Bonitasoft, 32 rue Gustave Eiffel - 38000 Grenoble
+ * This library is free software; you can redistribute it and/or modify it under the terms
+ * of the GNU Lesser General Public License as published by the Free Software Foundation
+ * version 2.1 of the License.
+ * This library is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth
+ * Floor, Boston, MA 02110-1301, USA.
+ **/
+
+package org.bonitasoft.engine.theme.impl;
+
+import java.security.NoSuchAlgorithmException;
+
+import org.bonitasoft.engine.commons.io.IOUtil;
+import org.bonitasoft.engine.log.technical.TechnicalLogSeverity;
+import org.bonitasoft.engine.log.technical.TechnicalLoggerService;
+import org.bonitasoft.engine.theme.model.STheme;
+
+/**
+ * @author Elias Ricken de Medeiros
+ */
+public class ThemeActionCalculator {
+
+    private final TechnicalLoggerService loggerService;
+
+    public ThemeActionCalculator(TechnicalLoggerService loggerService) {
+        this.loggerService = loggerService;
+    }
+
+    public enum ThemeAction {
+
+        CREATE,
+
+        UPDATE,
+
+        NONE
+
+    }
+
+    public ThemeAction calculateAction(STheme currentTheme, byte[] newContent) {
+        if(!hasContent(newContent)) {
+            return ThemeAction.NONE;
+        }
+        if (currentTheme == null) {
+            return ThemeAction.CREATE;
+        }
+        if (currentTheme.isDefault() && hasContentChanged(currentTheme, newContent)) {
+            return ThemeAction.UPDATE;
+        }
+        return ThemeAction.NONE;
+    }
+
+    private boolean hasContentChanged(final STheme currentTheme, final byte[] newContent) {
+        boolean changed = false;
+        try {
+            changed = !IOUtil.md5(currentTheme.getContent()).equals(IOUtil.md5(newContent));
+        } catch (NoSuchAlgorithmException e) {
```

Comment:
Actually this exception occurs when the algorithm MD5 is not available. It's not related to corrupted data.